lib/Controller/MentionController.php

```diff
@@ -41,22 +44,21 @@ public function mention(int $fileId, string $mention): DataResponse {
 			return new DataResponse([], Http::STATUS_NOT_FOUND);
 		}
 
-		// Reverse the array of users to pop off the first user later
-		$userResults = array_reverse($this->userManager->searchDisplayName($mention, 1));
-		if (count($userResults) < 1) {
+		[$searchResults, ] = $this->collaboratorSearch->search($mention, [IShare::TYPE_USER], false, 1, 0);
+		$matchedUsers = $searchResults['exact']['users'];
+		if (count($matchedUsers) < 1) {
 			return new DataResponse([], Http::STATUS_NOT_FOUND);
 		}
-
-		// Get the first user returned in the array
-		$user = array_pop($userResults);
-		$userFolder = $this->rootFolder->getUserFolder($user->getUID());
+		
+		$user = array_pop($matchedUsers);
+		$userFolder = $this->rootFolder->getUserFolder($user['shareWithDisplayNameUnique']);
 		$file = $userFolder->getFirstNodeById($fileId);
 		if ($file === null) {
 			return new DataResponse([], Http::STATUS_NOT_FOUND);
 		}
 
 		$notification = $this->manager->createNotification();
-		$notification->setUser($user->getUID())
+		$notification->setUser($user['shareWithDisplayNameUnique'])
```

Comment:
We need the user id to send the notification to the specific user, not the displayname, iirc it should be `$user['shareWith']`

Reply:
Thanks for catching that, I'll give it a fix

Reply:
An interesting thing I noticed when working on this is that, the `shareWithDisplayNameUnique` array value actually always points to the user ID, whereas the `label` value is what points to the actual non-unique display name. `shareWith` does in fact point to the user ID as well, so I'm not sure what the difference is. The notification gets sent just fine in either case -- is it preffered anyway to use `shareWith` as opposed to `shareWithDisplayNameUnique`? @juliusknorr

Reply:
@elzody your test users do not have email set, do they? Yes, please use `shareWith`.

Reply:
It makes sense now, thanks Arthur